Mathlib/NumberTheory/LSeries/Dirichlet.lean

```diff
@@ -0,0 +1,415 @@
+
+/-- A twisted version of the relation `Λ * ↑ζ = log` in terms of complex sequences. -/
+lemma convolution_twist_vonMangoldt {N : ℕ} (χ : DirichletCharacter ℂ N) :
+    (↗χ * ↗Λ) ⍟ ↗χ = ↗χ * ↗Complex.log := by
+  nth_rewrite 2 [← mul_one ↗χ]
+  rw [← DirichletCharacter.mul_convolution_distrib, convolution_vonMangoldt_const_one]
+
+/-- The L-series of the twist of the von Mangoldt function `Λ` by a Dirichlet character `χ`
+converges at `s` when `re s > 1`. -/
+lemma LSeriesSummable_twist_vonMangoldt {N : ℕ} (χ : DirichletCharacter ℂ N) {s : ℂ}
+    (hs : 1 < s.re) :
+    LSeriesSummable (↗χ * ↗Λ) s :=
+  LSeriesSummable_mul χ <| LSeriesSummable_vonMangoldt hs
+
+/-- The L-series of the twist of the von Mangoldt function `Λ` by a Dirichlet character `χ` at `s`
+equals the negative logarithmtic derivative of the L-series of `χ` when `re s > 1`. -/
+lemma LSeries_twist_vonMangoldt_eq {N : ℕ} (χ : DirichletCharacter ℂ N) {s : ℂ} (hs : 1 < s.re) :
+    L (↗χ * ↗Λ) s = - deriv (L ↗χ) s / L ↗χ s := by
+  rcases eq_or_ne N 0 with rfl | hN
+  · simp only [modZero_eq_delta, delta_mul_eq_smul_delta, vonMangoldt_apply_one, ofReal_zero,
+      zero_smul, LSeries_zero, Pi.zero_apply, LSeries_delta, Pi.one_apply, div_one, zero_eq_neg]
+    exact deriv_const s 1
+  -- now `N ≠ 0`
+  have hχ : LSeriesSummable ↗χ s := (LSeriesSummable_iff hN χ).mpr hs
+  have hs' : abscissaOfAbsConv ↗χ < s.re := by
+    rwa [absicssaOfAbsConv_eq_one hN, ← EReal.coe_one, EReal.coe_lt_coe_iff]
+  have hΛ : LSeriesSummable (↗χ * ↗Λ) s := LSeriesSummable_twist_vonMangoldt χ hs
+  rw [eq_div_iff <| LSeries_ne_zero_of_one_lt_re χ hs, ← LSeries_convolution' hΛ hχ,
+    convolution_twist_vonMangoldt, LSeries_deriv hs', neg_neg]
+  exact LSeries_congr s fun _ ↦ by simp only [Pi.mul_apply, mul_comm, logMul]
+
+end DirichletCharacter
+
+namespace ArithmeticFunction
+
+open DirichletCharacter in
+/-- The L-series of the von Mangoldt function `Λ` equals the negative logarithmic derivative
+of the L-series of the constant sequence `1` on its domain of convergence `re s > 1`. -/
+lemma LSeries_vonMangoldt_eq {s : ℂ} (hs : 1 < s.re) : L ↗Λ s = - deriv (L 1) s / L 1 s := by
+  refine (LSeries_congr s fun {n} _ ↦ ?_).trans <|
+    LSeries_modOne_eq ▸ LSeries_twist_vonMangoldt_eq χ₁ hs
+  simp only [Subsingleton.eq_one (n : ZMod 1), map_one, Pi.mul_apply, one_mul]
+
+/-- The L-series of the von Mangoldt function `Λ` equals the negative logarithmic derivative
+of the Riemann zeta function on its domain of convergence `re s > 1`. -/
+lemma LSeries_vonMangoldt_eq_deriv_riemannZeta_div {s : ℂ} (hs : 1 < s.re) :
+    L ↗Λ s = - deriv riemannZeta s / riemannZeta s := by
+  convert LSeries_vonMangoldt_eq hs using 2
+  · refine neg_inj.mpr <| Filter.EventuallyEq.deriv_eq <| Filter.eventuallyEq_iff_exists_mem.mpr ?_
+    exact ⟨{z | 1 < z.re}, (isOpen_lt continuous_const continuous_re).mem_nhds hs,
+      fun _ hz ↦ (LSeries_one_eq_riemannZeta hz).symm⟩
+  · exact (LSeries_one_eq_riemannZeta hs).symm
```

Comment:
`convert` is a slow tactic, the following is twice as fast:
```suggestion
lemma LSeries_vonMangoldt_eq_deriv_riemannZeta_div {s : ℂ} (hs : 1 < s.re) :
    L ↗Λ s = - deriv riemannZeta s / riemannZeta s := by
  suffices deriv (L 1) s = deriv riemannZeta s by
    rw [LSeries_vonMangoldt_eq hs, ← LSeries_one_eq_riemannZeta hs, this]
  refine Filter.EventuallyEq.deriv_eq <| Filter.eventuallyEq_iff_exists_mem.mpr ?_
  exact ⟨{z | 1 < z.re}, (isOpen_lt continuous_const continuous_re).mem_nhds hs,
    fun _ hz ↦ (LSeries_one_eq_riemannZeta hz)⟩```

Reply:
True; thanks!